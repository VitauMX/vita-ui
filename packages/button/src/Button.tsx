import * as React from 'react';
import styled from 'styled-components';
import { variant } from 'styled-system';
import buttonBaseStyles from './ButtonStyles';

const variants = variant({
  variants: {
    primary: {
      color: '#FFFFFF',
      borderWidth: '0px',
      padding: '0px 32px',
      fontFamily: 'Open Sans',
      backgroundColor: '#0099E6',
      '&:hover:enabled': {
        backgroundColor: '#0088CC',
      },
      '&:active:enabled': {
        backgroundColor: '#0077B3',
      },
      '&:focus:enabled': {
        boxShadow: '0px 0px 0px 2px #80D4FF',
      },
      '&:disabled': {
        cursor: 'not-allowed',
        backgroundColor: '#80D4FF',
      },
    },
  },
});

const destructive = variant({
  prop: 'destructive',
  variants: {
    true: {
      backgroundColor: '#FF2D1A',
      '&:hover:enabled': {
        backgroundColor: '#CC1100',
      },
      '&:active:enabled': {
        backgroundColor: '#B30F00',
      },
      '&:focus:enabled': {
        boxShadow: '0px 0px 0px 2px #FF8A7F',
      },
      '&:disabled': {
        cursor: 'not-allowed',
        backgroundColor: '#FF5B4C',
      },
    },
  },
});

const size = variant({
  prop: 'size',
  variants: {
    large: {
      height: '56px',
    },
    small: {
      padding: '0px 16px',
      height: '32px',
    },
  },
});

export interface IButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  variant: 'primary' | 'secondary';
  destructive?: boolean;
  size?: 'large' | 'small';
  disabled?: boolean;
}

const StyledButton = styled.button`
  ${buttonBaseStyles}
  ${variants}
  ${destructive}
  ${size}
`;

export const Button = React.forwardRef<HTMLButtonElement, IButtonProps>(
  (props, ref) => {
    return <StyledButton ref={ref} disabled={props.disabled} {...props} />;
  }
);

Button.displayName = 'Button';
