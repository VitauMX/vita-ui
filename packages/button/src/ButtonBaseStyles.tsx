import { css } from 'styled-components';

export default css`
  border: none;
  font-family: inherit;
  cursor: pointer;
  box-sizing: border-box;
  position: relative;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  margin: 0;
  border-radius: 4px;
  font-size: 1rem;
  line-height: 1;
  user-select: none;
  text-align: center;
  text-decoration: none;
  white-space: nowrap;
  appearance: none;

  transition: 220ms all ease-in-out;

  &:focus {
    outline: none;
  }

  &:disabled {
    cursor: default;
  }
`;
