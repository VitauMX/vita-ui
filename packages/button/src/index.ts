export * from './ButtonPrimary';
export * from './ButtonTertiary';
export * from './ButtonSecondary';
export * from './ButtonText';
