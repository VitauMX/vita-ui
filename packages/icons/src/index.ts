export * from './Icon';
export * from './createIcon';
export * from './icons/';
