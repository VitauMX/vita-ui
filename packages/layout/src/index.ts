export * from './Grid';
export * from './Box';
export * from './Flex';
