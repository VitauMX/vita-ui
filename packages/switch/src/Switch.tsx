import * as React from 'react';
import styled from 'styled-components';

export interface ISwitchProps {
  name: string;
}

const SwitchLabel = styled.label`
  position: absolute;
  width: 42px;
  height: 24px;
  border-radius: 22px;
  background: #bac4ca;
  cursor: pointer;
  &::after {
    content: '';
    display: block;
    border-radius: 50%;
    position: absolute;
    left: 5.36%;
    right: 48.21%;
    top: 9.38%;
    bottom: 9.38%;

    background: #ffffff;
    box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1);
    transition: 0.2s;
  }
`;

const SwitchBox = styled.input`
  display: none;
  &:checked + ${SwitchLabel} {
    background: #1ab3ff;
    &::after {
      left: 48.21%;
      right: 5.36%;
      top: 9.38%;
      bottom: 9.38%;

      background: #ffffff;
      box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1);
      transition: 0.2s;
    }
  }
`;

export const Switch = React.forwardRef<HTMLInputElement, ISwitchProps>(
  (props, ref) => (
    <>
      <SwitchBox id={props.name} type="checkbox" role="switch" ref={ref} />
      <SwitchLabel htmlFor={props.name} />
    </>
  )
);

Switch.displayName = 'Switch';
