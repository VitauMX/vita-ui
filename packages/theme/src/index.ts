export * from './ThemeProvider';
export * from './theme';
export * from './utils';
