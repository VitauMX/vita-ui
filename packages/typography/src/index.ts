export * from './Body';
export * from './Caption';
export * from './Label';
export * from './Message';
export * from './Title';
export * from './Overline';
