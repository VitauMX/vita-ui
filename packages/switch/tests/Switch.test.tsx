import React from 'react';
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';

import { Switch } from '../src';

describe('Switch', () => {
  test('renders correctly', () => {
    const { container } = render(<Switch name="test" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should have no axe violations', async () => {
    const { container } = render(<Switch name="test" />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
