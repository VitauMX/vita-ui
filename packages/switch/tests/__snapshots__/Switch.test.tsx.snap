// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch renders correctly 1`] = `
.c0 {
  display: none;
}

.c0:checked + .c1 {
  background: #1ab3ff;
}

.c0:checked + .c1::after {
  left: 48.21%;
  right: 5.36%;
  top: 9.38%;
  bottom: 9.38%;
  background: #ffffff;
  box-shadow: 0px 2px 4px rgba(0,0,0,0.1);
  -webkit-transition: 0.2s;
  transition: 0.2s;
}

<input
  class="c0"
  id="test"
  role="switch"
  type="checkbox"
/>
`;
